gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientLibraryWriteChannelTest.java

```diff
@@ -0,0 +1,293 @@
+            () -> {
+              writeStartedLatch.countDown();
+              try {
+                IOException ioe = assertThrows(IOException.class, writeChannel::close);
+                assertThat(ioe).isInstanceOf(ClosedByInterruptException.class);
+              } finally {
+                threadsDoneLatch.countDown();
+              }
+            });
+    // Wait for the insert object to be executed, then cancel the writing thread, and finally wait
+    // for the two threads to finish.
+    assertWithMessage("Neither thread started.")
+        .that(writeStartedLatch.await(5000, TimeUnit.MILLISECONDS))
+        .isTrue();
+    // callForClose will be waiting on write(waiting forever) to finish. Interrupt it.
+    callForClose.cancel(/* interrupt= */ true);
+    assertWithMessage("Failed to wait for tasks to get interrupted.")
+        .that(threadsDoneLatch.await(5000, TimeUnit.MILLISECONDS))
+        .isTrue();
+  }
+
+  private ByteString createTestData(int numBytes) {
+    byte[] result = new byte[numBytes];
+    for (int i = 0; i < numBytes; ++i) {
+      // Sequential data makes it easier to compare expected vs. actual in
+      // case of error. Since chunk sizes are multiples of 256, the modulo
+      // ensures chunks have different data.
+      result[i] = (byte) (i % 257);
+    }
+
+    return ByteString.copyFrom(result);
+  }
+
+  private GoogleCloudStorageClientLibraryWriteChannel getJavaStorageChannel() {
+    return new GoogleCloudStorageClientLibraryWriteChannel(
+        mockedStorage,
+        GoogleCloudStorageOptions.DEFAULT.toBuilder()
+            .setWriteChannelOptions(
+                AsyncWriteChannelOptions.DEFAULT.toBuilder().setGrpcChecksumsEnabled(true).build())
+            .build(),
+        resourceId,
+        CreateObjectOptions.DEFAULT_NO_OVERWRITE.toBuilder()
+            .setContentType(CONTENT_TYPE)
+            .setContentEncoding(CONTENT_ENCODING)
+            .setMetadata(getDecodedMetadata())
+            .setKmsKeyName(KMS_KEY)
+            .build(),
+        EXECUTOR_SERVICE);
+  }
+
+  private void verifyBlobInfoProperties() {
+    BlobInfo blobInfo = blobInfoCapture.getValue();
+    assertThat(blobInfo.getBucket()).isEqualTo(resourceId.getBucketName());
+    assertThat(blobInfo.getName()).isEqualTo(resourceId.getObjectName());
+    assertThat(blobInfo.getContentType()).isEqualTo(CONTENT_TYPE);
+    assertThat(blobInfo.getContentEncoding()).isEqualTo(CONTENT_ENCODING);
+    assertThat(blobInfo.getMetadata()).isEqualTo(metadata);
+  }
+
+  private void verifyBlobWriteOptionProperties() {
+    List<BlobWriteOption> optionsList = blobWriteOptionsCapture.getAllValues();
+    assertThat(optionsList).contains(BlobWriteOption.kmsKeyName(KMS_KEY));
+    assertThat(optionsList).contains(BlobWriteOption.generationMatch());
+    assertThat(optionsList).contains(BlobWriteOption.disableGzipContent());
+    assertThat(optionsList).contains(BlobWriteOption.crc32cMatch());
```

Comment:
Assert whole list content instead of each value individually.

Reply:
Updated.